server/src/test/java/org/elasticsearch/action/fieldcaps/FieldCapabilitiesFilterTests.java

```diff
@@ -19,6 +19,34 @@
 
 public class FieldCapabilitiesFilterTests extends MapperServiceTestCase {
 
+    /**
+     * Tests that the field family type is returned instead of the actual field type.
+     * `half_float` is chosen arbitrarily for this purpose.
```

Comment:
Instead of arbitrarily choosing `half_float` you could use a `randomFrom` and list there all the "float" types.